code/obj.dm

```diff
@@ -103,6 +103,11 @@
 		remove_dialogs()
 		..()
 
+	UpdateName()
+		if (src.real_name == null && name != null)
```

Comment:
`isnull` is the stricter check, for example `"" == null` is true but `isnull("")` is not.
```suggestion
		if (isnull(src.real_name && !isnull(src.name))
```